Libp2p K 8

mickvandijke opened the pull request.

duguorong009 committed:
fix: correct format of atto token decimals

duguorong009 committed:
Merge branch 'main' into fix-attos-decimal-format

RolandSherwin committed:
feat!: rename `home-network` flag to `relay`

RolandSherwin committed:
feat!: use `no-upnp` flag across the whole codebase

RolandSherwin committed:
feat!: rename `peers_args` into `initial_peers_config`

duguorong009 committed:
Merge branch 'main' into fix-attos-decimal-format

RolandSherwin committed:
feat(service_manager): impl versioning for `NodeServiceData`

RolandSherwin committed:
feat(test): add tests for `NodeServiceData` version deserialization

RolandSherwin committed:
fix(network): check if upnp is enabled via swarm

mickvandijke committed:
chore: patch libp2p to use fork with K = 8

mickvandijke committed:
refactor: get n closest peers instead of k closest

mickvandijke committed:
chore: update libp2p import

RolandSherwin committed:
feat: add logs for connection tracking

RolandSherwin committed:
feat: print error type on connection error

RolandSherwin committed:
chore: add docs for the changes in node service data v1

jacderida committed:
Merge branch 'rc-2025.4.1'

dependabot[bot] committed:
chore(deps): bump crossbeam-channel from 0.5.14 to 0.5.15

Bumps [crossbeam-channel](https://github.com/crossbeam-rs/crossbeam) from 0.5.14 to 0.5.15.
- [Release notes](https://github.com/crossbeam-rs/crossbeam/releases)
- [Changelog](https://github.com/crossbeam-rs/crossbeam/blob/master/CHANGELOG.md)

---
updated-dependencies:
- dependency-name: crossbeam-channel
  dependency-version: 0.5.15
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dirvine committed:
Update python-publish-client.yml

add rustfmt to musl build

maqi committed:
Merge pull request #2890 from maidsafe/dependabot/cargo/crossbeam-channel-0.5.15

chore(deps): bump crossbeam-channel from 0.5.14 to 0.5.15

dependabot[bot] committed:
chore(deps): bump tokio from 1.43.0 to 1.43.1

Bumps [tokio](https://github.com/tokio-rs/tokio) from 1.43.0 to 1.43.1.
- [Release notes](https://github.com/tokio-rs/tokio/releases)
- [Commits](https://github.com/tokio-rs/tokio/compare/tokio-1.43.0...tokio-1.43.1)

---
updated-dependencies:
- dependency-name: tokio
  dependency-version: 1.43.1
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

maqi committed:
Revert "test: enable data_with_churn using networkspawner"

This reverts commit bdd34293355d0e6f7b98e0901ba2da268483ed7b.

maqi committed:
Revert "test: use nodespawner for verify_data_location & verify_routing_table"

This reverts commit 90b02d8409dc0720536d6f670362e72c79a899ab.

jacderida committed:
Merge pull request #2918 from jacderida/chore-remove_ant_node_dev_dep

test: remove network test from autonomi crate

maqi committed:
Merge pull request #2884 from maidsafe/dependabot/cargo/tokio-1.43.1

chore(deps): bump tokio from 1.43.0 to 1.43.1

maqi committed:
Merge pull request #2922 from maqi/revert_use_nodespawner_in_ci_runs

Revert use nodespawner in ci runs

jacderida committed:
Merge pull request #2919 from dirvine/stable

Update python-publish-client.yml

mickvandijke committed:
fix(autonomi): set network id before fetching initial peers

maqi committed:
feat(network): disable disjoint_query_path

grumbach committed:
Merge pull request #2926 from mickvandijke/fix-client-set-network-id

fix(autonomi): set network id before fetching initial peers

maqi commented:
Closing this PR as the devnet run showing regression on resource usage.

Re-open if necessary.

mickvandijke commented:
Still waiting on a new devnet run.

grumbach committed:
feat: add cli analyzer docs

RolandSherwin committed:
fix(network): variance is now being calculated correctly

RolandSherwin committed:
chore: use open-metrics feature flag on kBucketStatus field

RolandSherwin committed:
fix(network): reduce the max bootstrap cache save interval

RolandSherwin committed:
fix(bootstrap_cache): do not remove expired addrs from the cache

grumbach committed:
feat: sync register key in vault

grumbach committed:
fix: prevent key overwrite during sync

grumbach committed:
fix: support old version of user data in order not to break existing vaults

maqi committed:
chore: reduce non-mandatory loggings

1. Reduce two node logs of GetVersion from Debug to Trace
2. Reduce one node log of Fetched Version from Info to Trace
3. Reduce one network log of Dial before Request from Info to Debug
4. Reduce two network logs of Circuit from Info to Debug

mickvandijke committed:
chore(evmlib): update alloy from `0.7.3` to `0.15.6`

jacderida committed:
docs: changelog for 2025.4.1.1 release

This is being added retrospectively because time did not permit it to be put together on the day of
the release.

jacderida committed:
Merge pull request #2939 from jacderida/docs-changelog_for_2025.4.1.1

docs: changelog for 2025.4.1.1 release

maqi committed:
chore: avoid extra escape backslash in json format

maqi committed:
feat(client): change quoting parallism to be linked to CHUNK_UPLOAD_BATCH_SIZE

maqi committed:
feat(client): extends client query_timout from 60s to 120s

maqi committed:
fix(client): retry shall be carried out if get_closest returns not enough entries

jacderida committed:
Merge pull request #2883 from RolandSherwin/network_flags_fix

feat!: add versioning to NodeServiceData (BREAKING CHANGE)

maqi committed:
feat(client): default FILE_UPLOAD_BATCH_SIZE to 1

grumbach committed:
chore: simplify code

grumbach committed:
fix: vault sync issues

grumbach committed:
Merge branch 'main' into fix_register_key_should_be_synced

grumbach committed:
Merge branch 'main' into analyzer-cli-docs

maqi committed:
fix(cli): refactor misleading paying_for printouts

mickvandijke committed:
Merge pull request #2937 from mickvandijke/chore-update-alloy

chore(evmlib): update alloy from `0.7.3` to `0.15.6`

maqi committed:
Merge pull request #2940 from maqi/avoid_extra_escape_backslash_in_json_format

chore: avoid extra escape backslash in json format

jacderida committed:
chore(release): stable release 2025.4.1.2

==================
  Crate Versions
==================
ant-bootstrap: 0.2.2
ant-build-info: 0.1.27
ant-cli: 0.3.13
ant-evm: 0.1.13
ant-logging: 0.2.49
ant-metrics: 0.1.28
ant-networking: 0.3.11
ant-node: 0.3.11
ant-node-manager: 0.12.3
ant-node-rpc-client: 0.6.46
ant-protocol: 1.0.4
ant-service-management: 0.4.12
ant-token-supplies: 0.1.66
autonomi: 0.4.5
evmlib: 0.3.0
evm-testnet: 0.1.13
nat-detection: 0.2.20
node-launchpad: 0.5.7
nodejs: 0.1.0
test-utils: 0.4.19
===================
  Binary Versions
===================
ant: 0.3.13
antctl: 0.12.3
antctld: 0.12.3
antnode: 0.3.11
antnode_rpc_client: 0.6.46
nat-detection: 0.2.20
node-launchpad: 0.5.7

jacderida committed:
docs: changelog for 2025.4.1.2 hotfix release

jacderida committed:
Merge pull request #2942 from maidsafe/rc-2025.4.1-hotfix1

Merge `2025.4.1.2` hotfix release to `main`

maqi committed:
Merge pull request #2931 from grumbach/analyzer-cli-docs

feat: add cli analyzer docs

maqi committed:
Merge pull request #2935 from grumbach/fix_register_key_should_be_synced

feat: sync register key in vault

maqi committed:
Merge pull request #2875 from duguorong009/fix-attos-decimal-format

fix: correct `AttoTokens` decimal format
